### PR TITLE
METRON-452: Add rudimentary configuration management functions to Stellar

### PR DESCRIPTION
**Pull request by cestella**
Currently, the only way to add enrichments is via editing JSON files and pushing them to zookeeper. Rather than that, because we have a REPL, we should create management functions around:
* Shell functions - Functions surrounding interacting with the shell in either a nicer way or a more functional way.
* Configuration functions - Functions surrounding pulling and pushing configs from zookeeper
* Parser functions - Functions surrounding adding, viewing, and removing Parser functions.
* Enrichment functions - Functions surrounding adding, viewing and removing Stellar enrichments as well as managing batch size and index names for the enrichment topology configuration
* Threat Triage functions - Functions surrounding adding, viewing and removing threat triage functions.

### CODE REVIEW

**Comment by cestella** [comment]
I have provided some fairly full-featured examples of using these functions within the REPL in the form of REPL transcripts [here](https://github.com/cestella/incubator-metron/blob/631b1a4adf416da8d9afc01f5798b67cf4c71934/metron-platform/metron-management/README.md#examples) in the docs as an aid to both understanding and testing.

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      FieldTransformer stellarTransformer = getStellarTransformer(configObj);
+      List<String> removals = (List<String>)args.get(1);
+      if(removals == null || removals.isEmpty()) {
+        return config;
+      }
+      for(String removal : removals) {
+        stellarTransformer.getConfig().remove(removal);
+      }
+      List<String> output = new ArrayList<>();
+      for(String key : stellarTransformer.getConfig().keySet()) {
```
Comment (justinleet) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java]: Can we make this `output.addAll(stellarTransformer.getConfig().keySet());`

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      FieldTransformer stellarTransformer = getStellarTransformer(configObj);
+      Map<String, String> additionalTransforms = (Map<String, String>) args.get(1);
+      if(additionalTransforms == null || additionalTransforms.isEmpty()) {
+        return config;
+      }
+      for(Map.Entry<String, String> kv : additionalTransforms.entrySet()) {
+        stellarTransformer.getConfig().put(kv.getKey(), kv.getValue());
+
+      }
+      List<String> output = new ArrayList<>();
```
Comment (justinleet) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java]: Same deal with `output.addAll(stellarTransformer.getConfig().keySet());`

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/EnrichmentConfigFunctions.java [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+  }
+  public static Map<String, Object> getStellarHandler(EnrichmentConfig enrichmentConfig) {
+    Map<String, Object> fieldMap = enrichmentConfig.getFieldMap();
+    Map<String, Object> stellarHandler = (Map<String, Object>) fieldMap.get("stellar");
+    if(stellarHandler == null ) {
+      stellarHandler = new HashMap<String, Object>();
+      fieldMap.put("stellar", stellarHandler);
+    }
+
+    if(stellarHandler.get("config") == null){
```
Comment (justinleet) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/EnrichmentConfigFunctions.java]: Can we use `stellarHandler.putIfAbsent("config", new LinkedHashMap<String, Object>());`?  It's a Java 8 thing, and I haven't seen anyone suggest moving back to Java 7, so might as well get some use out of those functions.

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+import static org.apache.metron.common.configuration.ConfigurationType.PARSER;
+
+public class ParserConfigFunctions {
+  private static final Logger LOG = Logger.getLogger(ConfigurationFunctions.class);
+
+  private static void pruneEmptyStellarTransformers(SensorParserConfig config) {
+    List<FieldTransformer> toRemove = new ArrayList<>();
+    List<FieldTransformer> fieldTransformations = config.getFieldTransformations();
+    for(FieldTransformer transformer : fieldTransformations) {
+      if(transformer.getFieldTransformation().getClass().getName()
```
Comment (cestella) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java]: made statements less questionable

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java (hunk 1 above)
Comment (cestella) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/ParserConfigFunctions.java]: done

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/EnrichmentConfigFunctions.java (hunk 3 above)
Comment (cestella) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/EnrichmentConfigFunctions.java]: done

**Comment by justinleet** [comment]
I'm +1, pending the Travis check

**Pull request closed without merging**